See https://sourceforge.net/p/openil/patches/49/ -- this patch allows
DevIL to be configured with ILUT, but without X11:

--- m4/devil-api_checks.m4	2009-03-08 10:10:05.000000000 +0300
+++ m4/devil-api_checks.m4	2012-01-31 17:05:07.163553444 +0400
@@ -137,7 +137,7 @@
                               [main],
                               [dnl The use_x11 var is either "yes" already, or we don't want "no" to be overwritten
                                ILUT_LIBS="-lX11 $ILUT_LIBS"],
-                              [use_x11="no"]) ])
+                              [use_x11="no"]) ], [use_x11="no"])
           TEST_API([shm])
           AS_IF([test "x$enable_shm" = "xyes"],
                 [AC_CHECK_HEADER([X11/extensions/XShm.h],
